src/Microsoft.ML.Transforms/SvmLight/SvmLightLoaderSaverCatalog.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Linq;
+using Microsoft.ML.Data;
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML.Transforms
+{
+    public static class SvmLightLoaderSaverCatalog
+    {
+        /// <summary>
+        /// Creates a loader that loads SVM-light format files. <see cref="SvmLightLoader"/>.
+        /// </summary>
+        /// <param name="catalog">The <see cref="DataOperationsCatalog"/> catalog.</param>
+        /// <param name="inputSize">The number of features in the Features column. If 0 is specified, the
+        /// loader will determine it by looking at the file sample given in <paramref name="dataSample"/>.</param>
+        /// <param name="numberOfRows">The number of rows from the sample to be used for determining the number of features.</param>
+        /// <param name="zeroBased">If the file contains zero-based indices, this parameter should be set to true. If they are one-based
+        /// it should be set to false.</param>
+        /// <param name="dataSample">A data sample to be used for determining the number of features in the Features column.</param>
+        public static SvmLightLoader CreateSvmLightLoader(this DataOperationsCatalog catalog,
+            int inputSize = 0,
+            long? numberOfRows = null,
+            bool zeroBased = false,
+            IMultiStreamSource dataSample = null)
+            => new SvmLightLoader(CatalogUtils.GetEnvironment(catalog), new SvmLightLoader.Options()
+            { InputSize = inputSize, NumberOfRows = numberOfRows, FeatureIndices = zeroBased ?
+                SvmLightLoader.FeatureIndices.ZeroBased : SvmLightLoader.FeatureIndices.OneBased }, dataSample);
+
+        /// <summary>
+        /// Creates a loader that loads SVM-light like files, where features are specified by their names.
+        /// </summary>
+        /// <param name="catalog">The <see cref="DataOperationsCatalog"/> catalog.</param>
+        /// <param name="numberOfRows">The number of rows from the sample to be used for determining the set of feature names.</param>
+        /// <param name="dataSample">A data sample to be used for determining the set of features names.</param>
+        public static SvmLightLoader CreateSvmLightLoaderWithFeatureNames(this DataOperationsCatalog catalog,
+            long? numberOfRows = null,
```

Comment:
CreateTextLoader two versions with different parameters. Is that a better practice to follow and call both functions CreateSvmLightLoader?

Reply:
Do you think I should have one method that takes an `SvmLightLoader.FeatureIndices` as a parameter?
In that case, perhaps we could come up with a better name for that enum? Any suggestions?

---
In reply to: [332206307](https://github.com/dotnet/machinelearning/pull/4190#discussion_r332206307) [](ancestors = 332206307)

Reply:
Yes, I think having a method that takes FeatureIndices as a parameter would be helpful to distinguish the two cases. One option would be to just call the enum as `Features` and call the enum values as `ZeroBasedIndices`, `OneBaseIndices` and `Names`. 
Also, both the methods aboe have numberOfRows as a parameter, but they are in different positions. If the two functions have the same parameter, it would be good to keep them in the same position. 


---
In reply to: [333653705](https://github.com/dotnet/machinelearning/pull/4190#discussion_r333653705) [](ancestors = 333653705,332206307)

Reply:
I've addressed the parameter-order part of your comment, but I had second thoughts about having a single API that specifies whether the file contains feature indices or feature names - there is a parameter `inputSize` which is only relevant in the feature indices case and not in the feature names case, and I'm not sure that having a parameter that is sometimes ignored is the best thing to have - do you have any thoughts on that?

---
In reply to: [337250107](https://github.com/dotnet/machinelearning/pull/4190#discussion_r337250107) [](ancestors = 337250107,333653705,332206307)

Reply:
Based on your explanation, would reordering and renaming the params as follows work?
```
        // For namesCreateSvmLightLoader to use with indices
        public static SvmLightLoader CreateSvmLightLoader(
            this DataOperationsCatalog catalog,
            long? numberOfRows = null,
            IMultiStreamSource dataSample = null,
            Features features,
            int featureSize,
            );

        // For namesCreateSvmLightLoaderWithNames
        public static SvmLightLoader CreateSvmLightLoader( 
            this DataOperationsCatalog catalog,
            long? numberOfRows = null,
            IMultiStreamSource dataSample = null,
            Features features
            );
```
The other option is to keep only one version, (whichever you think would be the more common one and change the other one to have the SvmLightLoader.Options class directly as a param. That too has been a pattern in ML.NET.

In either case I think input from @eerhardt would be useful for the public API surface.



---
In reply to: [343611210](https://github.com/dotnet/machinelearning/pull/4190#discussion_r343611210) [](ancestors = 343611210,337250107,333653705,332206307)